.github/workflows/osx_build.yml

```diff
@@ -0,0 +1,52 @@
+name: MacOS build and test rolesanywhere-credential-helper
+on:
+  push:
+    branches: [ main ]
+  
+permissions:
+  statuses: write
+  id-token: write
+
+jobs:
+  OSX:
+    name: Build on ${{matrix.runner_image}}
+    runs-on: ${{matrix.runner_image}}
+    strategy:
+      matrix:
+        include:
+          - runner_image: macos-13
+            s3_path: X86_64/OSX/MacOS
+            display_name: "MacOS Intel"
+          - runner_image: macos-14
+            s3_path: Aarch64/OSX/MacOS
+            display_name: "MacOS Arm64"
+
+    steps: 
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Setup go
+        uses: actions/setup-go@v5
+        with:
+          go-version-file: 'go.mod'
+          cache: true
+
+      - name: Build Project
+        run: |
+          echo "Building on ${{ matrix.display_name }}"
+          make release
+          make test-certs
+          make test
```

Comment:
The `test` target already depends on the `test-certs` target, so there's no need to run `test-certs` first explicitly. 

Also, have you been able to test negative scenarios here? If a unit test fails, does that cause the last command to return a non-zero exit code, and that in turn will make it so that the rest of the steps don't run?